test/parallel/test-http-socket-error-listeners.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const net = require('net');
+
+// This test sends an invalid character to a HTTP server and purposely
+// does not handle clientError (even if it sets an event handler).
+//
+// The idea is to let the server emit multiple errors on the socket,
+// mostly due to parsing error, and make sure they don't result
+// in leaking event listeners.
+
+{
+  let i = 0;
+  let socket;
+  const mustNotCall = common.mustNotCall.bind(null);
+  process.on('warning', mustNotCall);
+
+  const server = http.createServer(common.mustNotCall());
+
+  server.on('clientError', common.mustCallAtLeast((err) => {
+    assert.strictEqual(err.code, 'HPE_INVALID_METHOD');
+    assert.strictEqual(err.rawPacket.toString(), '*');
+
+    if (i === 20) {
+      socket.end();
+    } else {
+      socket.write('*');
+      i++;
+    }
+  }, 1));
+
+  server.listen(0, () => {
+    socket = net.createConnection({ port: server.address().port });
+
+    socket.on('connect', () => {
+      socket.write('*');
+    });
+
+    socket.on('close', () => {
+      server.close();
+      process.removeListener('warning', mustNotCall);
+    });
+  });
+}
+
+{
+  const valid = 'GET / HTTP/1.1\r\nHost:a\r\nContent-Length:1\r\n\r\n1\r\n';
+
+  const server = http.createServer((req, res) => {
+    const handler = common.mustCall(function handler(error) {
+      req.socket.removeListener('error', handler);
+    }, 2);
+
+    req.on('end', handler);
+    req.socket.on('error', handler);
```

Comment:
This is wrong and it is not a Node.js core issue.

Reply:
See https://github.com/nodejs/node/pull/46523#discussion_r1097881390